airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -1179,76 +1155,27 @@ def compare_records(
                     not expected_but_not_found
                 ), f"Expected to see those primary keys in the actual response for stream {stream_name} but they were not found."
         else:
-            TestBasicRead.legacy_compare_records(
-                stream_name, actual, expected, extra_fields, exact_order, extra_records, ignored_fields, detailed_logger
-            )
+            if len(expected) > len(actual):
```

Comment:
nit: It looks like the rest of the logging and fail code is all encompassed in this `if` conditional. Can we just combine the above `else` into `elif len(expected > len(actual)`

Reply:
You are right. In my mind, it was "if primary_key: do X, else: do Y" but the flow makes more sense as you describe